### PR TITLE
P1989R2 Range constructor for std::string_view 2: Constrain Harder

### LINKED ISSUE #4649: [2021-06 LWG Motion 12] P1989R2 Range constructor for std::string_view 2: Constrain Harder
[P1989R2](https://wg21.link/p1989r2)

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #4649
Fixes cplusplus/papers#714

### CODE REVIEW

**Comment by jensmaurer** [comment]
See http://lists.isocpp.org/lib/2021/06/19670.php for a glitch in the incoming paper.

@tkoeppe , should this be fixed editorially?

File: source/strings.tex [hunk 1]
```diff
@@ -4180,6 +4182,9 @@
   // \ref{string.view.deduct}, deduction guide
```
Comment (JohelEGP) [inline source/strings.tex:4182]:
```suggestion
  // \ref{string.view.deduct}, deduction guides
```

File: source/strings.tex [hunk 2]
```diff
@@ -4180,6 +4182,9 @@
...
   template<class It, class End>
     basic_string_view(It, End) -> basic_string_view<iter_value_t<It>>;
+
```
Comment (JohelEGP) [inline source/strings.tex:4185]:
```suggestion
```

File: source/strings.tex [hunk 3]
```diff
@@ -4279,6 +4284,44 @@
...
+\begin{itemdescr}
+\pnum
+Let \tcode{d} be an lvalue of type \tcode{remove_cvref_t<R>}.
+
+\pnum
+\constraints
+\begin{itemize}
+\item
+\tcode{R} models
+\tcode{ranges::contiguous_range} and \tcode{ranges::sized_range},
```
Comment (JohelEGP) [inline source/strings.tex:4302]:
```suggestion
\tcode{ranges::\libconcept{contiguous_range}} and \tcode{ranges::\libconcept{sized_range}},
```

File: source/strings.tex [hunk 4]
```diff
@@ -4970,6 +5013,17 @@
...
 
+\begin{itemdecl}
+template<class R>
+  basic_string_view(R&&) -> basic_string_view<ranges::range_value_t<R>>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constraints
+\tcode{R} satisfies \tcode{ranges::contiguous_range}.
```
Comment (JohelEGP) [inline source/strings.tex:5024]:
```suggestion
\tcode{R} satisfies \tcode{ranges::\libconcept{contiguous_range}}.
```

**Review by JohelEGP — CHANGES_REQUESTED** [review]

**Comment by jensmaurer** [comment]
@JohelEGP , thanks, fixed.

**Comment by tkoeppe** [comment]
@jensmaurer: yes, I'll assume that's fine and will add a fix. I've asked for LWG approval (@jwakely?), but will proceed assuming we'll get that.

**Pull request merged**